### DISCUSSION TITLE
The linear damping of PxRigidBody

### DISCUSSION THREAD

**Discussion by SHiziw** [NVIDIA-Omniverse/PhysX#241] [Q&A] [answered]
Hello NVIDIA Devs,

I have adjusted the damping of a PxRigidBody to 0.01 using the `setLinearDamping` method. Interestingly, I've observed that the maximum velocity induced by gravity remains constant, regardless of the mass of the object. However, according to Newton's second law:

$$mg - c\frac{dx}{dt}  = m\frac{d^2x}{dt^2}$$

we can derive that the maximum velocity is given by:

$$v_{max} = \frac{m g}{c}$$

This suggests that the maximum velocity should indeed be related to the mass of the object. I'm puzzled as to why this isn't the case in my observations.

Do you have any thoughts on why this is happening?

Thanks!

**Comment by preist-nvidia** [comment, marked as answer]
Hi @SHiziw,

Linear damping does not apply a force proportional to velocity, but scales the (unconstrained, pre-integrated) velocities each time step with

`max(0, 1 - linear_damping * dt)`

see, for example, here: https://github.com/NVIDIA-Omniverse/PhysX/blob/main/physx/source/lowleveldynamics/src/DyBodyCoreIntegrator.h#L62

I agree that the documentation does not make this clear enough, I'll update it.
Philipp

**Reply by SHiziw** [reply]
Thank you the detailed answer!
